### PR TITLE
(CI): add bytecode size and gas difference reporting

### PR DESCRIPTION
**Pull request by roleengineer**
Added gas tracking and bytecode size reporting to test workflow.

Integrated two new commands into the test workflow:

- ```forge build --sizes```: Prints bytecode sizes.

- ```forge snapshot --diff```: Prints gas difference between snapshots.

### CODE REVIEW

**Comment by roleengineer** [comment]
fixes https://github.com/aboutcircles/circles-contracts-v2/issues/96

File: test/hub/MockDeployment.sol [hunk 1]
```diff
@@ -14,6 +14,7 @@ import "../lift/MockERC20Lift.sol";
 import "./MockHub.sol";
 
 contract MockDeployment {
+    bool public IS_TEST = true;
```
Comment (benjaminbollen) [inline test/hub/MockDeployment.sol:17]: is `IS_TEST` to not count this for the gas measurement?

Comment (benjaminbollen) [reply test/hub/MockDeployment.sol:17]: sorry understood. it is for the build size

**Review by benjaminbollen — APPROVED** [review]
LGTM. Up next, we can look at frameworks to store the results and report them; add it to the readme

**Pull request merged**